### DISCUSSION TITLE
Browser Support?

### DISCUSSION THREAD

**Discussion by pusolito** [grpc/grpc-kotlin#442] [Q&A]
Would the generated client work in Kotlin JS? If not, are there plans to support this? I know about the gRPC-web project, but that seems focused on generating TypeScript/JavaScript. And while it's possible to leverage JS types directly from Kotlin JS, it would be nice to not need the extra complexity of declaring externals etc to use the pure TS/JS client.

**Comment by jamesward** [comment]
At this time Kotlin/JS isn't supported.  In the meantime, this project might support it: https://github.com/open-toast/protokt

**Reply by pusolito** [reply]
Thanks for the pointer! I'll take a look at protokt as well.

Is future support for Kotlin JS likely for this project though?

**Reply by jamesward** [reply]
I don't think it is likely.  Not impossible but if I were to put my money on something here it'd likely be Kotlin/Wasm, not Kotlin/JS.  No commitments or plans either way at this point.
